### ISSUE TITLE
Multi-byte factor names caused an error in ANOVA and ANCOVA in windows

### ISSUE THREAD

**Issue by sbtseiji** [jamovi/jmv#309]
In windows, jamovi 1.8.4's ANOVA and ANCOVA get an error when the variables in the fixed factors or the covariates have multibyte names (this causes no problem with the mac version). I tried with the jmv package in R and got the following error. 

> Error in nchar(elem) : invalid multibyte string, element 1

<img width="519" alt="screenshot" src="https://user-images.githubusercontent.com/8280906/124346423-b2fab500-dc19-11eb-8304-b428ff953f13.png">

Maybe it has something to do with the processing in utils.R in the jmvcore package?

**Comment by jonathon-love** [comment]
yeah, unicode handling in R, under windows, can be quite problematic.

could you try sharing a data set we can test this against?

with thanks

**Comment by sbtseiji** [comment]
thank you for a prompt reply. here is the data file in csv format.
[anova_data02.csv](https://github.com/jamovi/jmv/files/6758030/anova_data02.csv)

the second and the third columns are the fixed factor variables and the last column contains the dependent variable.

**Comment by jonathon-love** [comment]
hi,

perhaps somewhat predictably, it works fine here ... could you see if any of these work?

```
jmv::ANOVA(data, formula=得点~条件)

jmv::ANOVA(data, dep='得点', factors='条件')

dep = stringi::stri_enc_toutf8('得点')
factors = stringi::stri_enc_toutf8('条件')

jmv::ANOVA(data, dep=!!dep, factors=!!factors)
```

with thanks

**Comment by sbtseiji** [comment]
sure, 

I checked them with both the jmv module loaded to R 4.1.0 and jamovi's Rj module on Windows 10.

Here are the results.

with jmv package in R

`jmv::ANOVA(mydat, formula=得点~条件)`

works fine. results and the variable names were displayed properly.

`jmv::ANOVA(mydat, dep='得点',factors='条件')`

doesn't work. gave me back the error

`nchar(elem) でエラー: invalid multibyte string, element 1`

```
dep=stringi::stri_enc_toutf8('得点')
factors=stringi::stri_enc_toutf8('条件')
jmv::ANOVA(mydat, dep=!!dep, factors=!!factors)
```

works fine. results and the variable names were displayed properly.
 

in Rj module: 

`jmv::ANOVA(mydat, formula=得点~条件)`

doesn't work. gave me the error

`<text>:1:26: unexpected '<' 1: jmv::ANOVA(data, formula=< ^`

![sc1](https://user-images.githubusercontent.com/8280906/124439178-0f91d780-ddb4-11eb-8352-82dd845726af.png)


`jmv::ANOVA(data, dep='得点', factors = '条件')`

somehow works. results were calculated properly, but the variable names were displayed as a series of unicodes like <U+6761><U+7089>.

```
dep = stringi::stri_enc_toutf8('得点')
factors = stringi::stri_enc_toutf8('条件')
jmv::ANOVA(data, dep=!!dep, factors=!!factors)
```

same as above. results were calculated properly, but the variable names were displayed as a series of unicodes.

![sc3](https://user-images.githubusercontent.com/8280906/124439099-f4bf6300-ddb3-11eb-8293-4dff57793ee7.png)

**Comment by jonathon-love** [comment]
thanks. one last question, what does:

```
Encoding('得点')
```

return?

**Comment by sbtseiji** [comment]
OK, I tested it and got "unknown" in both Rj Editor and R4.1.0 console.

**Comment by jonathon-love** [comment]
hmm ... one more ...

```
Encoding(stringi::stri_enc_tonative('得点'))
```

oh, and maybe check that this still looks right:

```
stringi::stri_enc_tonative('得点')
```

with thanks

**Comment by sbtseiji** [comment]
Here are the results I've got. I also tested with `stringi::stri_enc_toutf8('得点')`

in Rj Editor

```
Encoding(stringi::stri_enc_tonative('得点'))
[1] "unknown"
stringi::stri_enc_tonative('得点')
[1] "<U+5F97><U+7089>"
Encoding(stringi::stri_enc_toutf8('得点'))
[1] "unknown"
stringi::stri_enc_toutf8('得点')
[1] "<U+5F97><U+7089>"
```

I think it's weird that I get unknown even when I explicitly set it to utf-8.

in R 4.1.0 console

```
> Encoding(stringi::stri_enc_tonative('得点'))
[1] "unknown"
> stringi::stri_enc_tonative('得点')
[1] "得点"
> Encoding(stringi::stri_enc_toutf8('得点'))
[1] "UTF-8"
> stringi::stri_enc_toutf8('得点')
[1] "得点"
```

In this case, Encoding() gives me back "UTF-8" as long as I encode the string to utf-8.

**Comment by jonathon-love** [comment]
> I think it's weird that I get unknown even when I explicitly set it to utf-8.

yeah, unicode handling in R is a real mess :/
